src/ensembl/src/shared/components/upload/Upload.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from 'react';
+import get from 'lodash/get';
+import forEach from 'lodash/forEach';
+import classNames from 'classnames';
+
+import Overlay from 'src/shared/components/overlay/Overlay';
+
+import styles from './Upload.scss';
+type PropsForRespondingWithASingleFile = {
+  onChange: (file: File) => void;
+  callbackWithFiles: true;
+  allowMultiple: false;
+};
+
+type PropsForRespondingWithMultipleFiles = {
+  onChange: (files: FileList) => void;
+  callbackWithFiles: true;
+  allowMultiple: true;
+};
+
+type PropsForRespondingWithContent = {
+  onChange: (content: string) => void;
+  callbackWithFiles: false;
+  allowMultiple: true;
+};
+
+type OnChangeProps =
+  | PropsForRespondingWithASingleFile
+  | PropsForRespondingWithMultipleFiles
+  | PropsForRespondingWithContent;
+
+export type UploadProps = {
+  id: string;
+  name?: string;
+  label: string;
+  className?: string;
+} & OnChangeProps;
+
+const Upload = (props: UploadProps) => {
+  const [drag, setDrag] = useState(false);
+
+  const fileReaders: FileReader[] = [];
+  let totalPendingFilesToRead = 0;
+  let dragCounter = 0;
+
+  const handleDragIn = (e: React.DragEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    dragCounter++;
+    if (get(e, 'dataTransfer.items.length', 0) > 0) {
+      setDrag(true);
+    }
+  };
+
+  const handleDragOut = (e: React.DragEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    dragCounter--;
+    if (dragCounter === 0) {
+      setDrag(false);
+    }
+  };
+
+  const handleDrop = (e: React.DragEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    setDrag(false);
+
+    if (
+      e.dataTransfer &&
+      e.dataTransfer.files &&
+      e.dataTransfer.files.length > 0
+    ) {
+      if (props.callbackWithFiles) {
+        if (!props.allowMultiple) {
+          props.onChange(e.dataTransfer.files[0]);
+          return;
+        }
+        props.onChange(e.dataTransfer.files);
+        return;
+      }
+
+      const { files } = e.dataTransfer;
+
+      forEach(files, (file) => {
```

Comment:
Why is the code using lodash's forEach instead of array’s native forEach? Does the `files` object not have a forEach? If not, can it be spread in an array (`[...files].forEach`)?

Reply:
`[...files].forEach` works 👍